packages/libs/kadena.js/src/util/PactNumber.ts

```diff
@@ -0,0 +1,50 @@
+import { IPactDecimal, IPactInt } from '@kadena/types';
+
+import BigNumber from 'bignumber.js';
+
+export class PactNumber {
+  private readonly _number: BigNumber;
+
+  public constructor(value: string | number) {
+    if (isNaN(Number(value))) throw new Error('Value is NaN');
+    this._number = new BigNumber(value);
+  }
+
+  public toInteger(): string {
+    if (!this._number.isInteger()) {
+      throw new Error('PactNumber is not an integer');
+    }
+    return this._number.toString();
+  }
+
+  public toStringifiedInteger(): string {
+    return JSON.stringify(this.toInteger());
+  }
+
+  public toPactInteger(): IPactInt {
+    if (!this._number.isInteger()) {
+      throw new Error('PactNumber is not an integer');
+    }
+    return {
+      int: this.toInteger(),
+    };
+  }
+
+  public toDecimal(): string {
+    if (this._number.isInteger()) {
+      return `${this._number.toString()}.0`;
+    }
+    console.log(this._number, this._number.isInteger());
+    return `${this._number.toString()}`;
+  }
+
+  public toStringifiedDecimal(): string {
+    return JSON.stringify(this.toDecimal());
+  }
+
+  public toPactDecimal(): IPactDecimal {
+    return {
+      decimal: this.toDecimal(),
+    };
+  }
+}
```

Comment:
Should we move it to a separate lib? We have two categories, low level libs, and consumer libs. This could be part of a low-level lib?